### PR TITLE
router: stop iOS subprocess sandbox violations

### LINKED ISSUE #597: ios IPNExtension sandbox violations trying to fork a subprocess
We now have sandbox violations in the iOS app as of recently. (unrelated version)

@dshynkev suspects 30bbbe9467 is what introduced it.

Filing bug so I don't forget before the weekend. :)

### PR DESCRIPTION
**Pull request by dshynkev**
The sandbox violations were due to `ifconfig down` being unintentionally included in `Close` for both darwin and freebsd. This fixes that. `downDNS` is a no-op on darwin, so it's fine to run it on both platforms. At the same time, `ifconfig down` is not actually the right thing to do on `freebsd`, as the command elaborates. Note that `cleanup` never runs in the darwin sandbox, as it is only called from `cmd/tailscaled`.

This is all a bit annoying to reason about; #615 refactors things into a better shape, but this is a minimal fix.

Resolves #597.

---
This change is [Reviewable](https://reviewable.io/reviews/tailscale/tailscale/617)

### CODE REVIEW

File: wgengine/router/router_userspace_bsd.go [hunk 1]
```diff
@@ -153,17 +153,23 @@ func (r *userspaceBSDRouter) Set(cfg *Config) error {
 }
 
 func (r *userspaceBSDRouter) Close() error {
-	cleanup(r.logf, r.tunname)
+	if err := downDNS(interfaceName); err != nil {
+		logf("dns down: %v", err)
```
Comment (bradfitz) [inline wgengine/router/router_userspace_bsd.go:157]: does this compile? where is logf?

**Comment by bradfitz** [comment]
Btw, it's not Darwin, but iOS.

File: wgengine/router/router_userspace_bsd.go (hunk 1 above)
Comment (dshynkev) [reply wgengine/router/router_userspace_bsd.go]: Let me quickly test this on an actual FreeBSD machine; I just adapted it from #615.

**Comment by dshynkev** [comment]
It certainly builds and runs. I cannot reproduce the sandbox violations before this change as I do not own an iOS device and the Xcode Simulator does not seem to emulate the sandbox; please let me know if it should.

File: wgengine/router/router_userspace_bsd.go [hunk 2]
```diff
@@ -153,17 +153,23 @@ func (r *userspaceBSDRouter) Set(cfg *Config) error {
...
 	if err := downDNS(interfaceName); err != nil {
 		logf("dns down: %v", err)
 	}
-
-	ifup := []string{"ifconfig", interfaceName, "down"}
+	// If the interface was left behind, ifconfig down will not remove it.
+	// In fact, this will leave a system in a tainted state where starting tailscaled
+	// will result in "interface tailscale0 already exists"
+	// until the defunct interface is ifconfig-destroyed.
+	ifup := []string{"ifconfig", interfaceName, "destroy"}
```
[the remark this answers is not in the document]
Comment (dshynkev) [reply wgengine/router/router_userspace_bsd.go:171]: At least on FreeBSD 12.1 on DigitalOcean, it does seem to be necessary. I was quite surprised myself.

**Pull request merged**